### ISSUE TITLE
building a HeatmapAnnotation only object by concatenating both horizontally and vertically HeatmapAnnotation objects

### ISSUE THREAD

**Issue by alevax** [jokergoo/ComplexHeatmap#459]
Is it possible to create an object to plot by concatenating both horizontally and vertically HeatmapAnnotation objects?

e.g.
```
ht_list = NULL
  for(i in 1:length(a.list) ) {
  ht_list = ht_list %v% (a.list[[i]] + b.list[[i]])
}
```

where both a.list and b.list are a lists of rowAnnotations.

**Comment by jokergoo** [comment]
No, but if you vertically align the heatmaps, horizontally, you need to set other heatmaps/annotations as left and right annotation. Similarly, if you horizontally align the heatmaps, the vertically aligned heatmaps/annotations need to be set as top/bottom annotations.

See the last plot in this section: https://jokergoo.github.io/ComplexHeatmap-reference/book/a-list-of-heatmaps.html#vertical-concatenation

<img width="739" alt="Screenshot 2020-04-10 at 09 56 45" src="https://user-images.githubusercontent.com/449218/78974108-96a0c300-7b11-11ea-9ee8-cf7647c1481e.png">

**Comment by alevax** [comment]
But, I am not using Heatmap, only HeatmapAnnotation. Does it work too? I don’t have Heatmap to use in this object composition.

**Comment by jokergoo** [comment]
But how can you align to a HeatmapAnnotation vertically and at the same time horizontally? Can you change one HeatmapAnnotation to a Heatmap? Character matrix also works for Heatmap.

**Comment by alevax** [comment]
To answer your first question, I tried to align both ways and didn't work out. I got an error message saying that apparently we cannot. To answer your second question, I successfully converted my HeatmapAnnotation (_anno_barplot_) to a Heatmap, but I cannot specify the heatmap cell/tile width, thing that I was able to do with _anno_barplot_ using the parameter **bar_width**. 

Is there any way to overcome this?
